utils/docker/vcluster/README.md

```diff
@@ -0,0 +1,197 @@
+# Using DAOS with Docker
+
+This section describes how to build and deploy Docker images allowing to simulate a small cluster
+using DAOS as backend storage.  This small cluster is composed of the following three nodes:
+
+- The `daos-server` node running a DAOS server daemon managing data storage devices such as SCM or
+  NVMe disks.
+- The `daos-admin` node allowing to manage the DAOS server thanks to `dmg`command.
+- The `daos-client` node using the the DAOS server to store data.
+
+At this time only emulated hardware storage are supported by this Docker platform:
+
+- SCM (i.e. Stoarage Class Memory) are emulated with standard RAM memory.
+- NVMe disks are emulated with a file device.
+
+!!! warning
+    Virtual Docker network such as [bridge](https://docs.docker.com/network/bridge/) are not yet
+    well supported by DAOS.  Thus, one physical network interface of the host should be choosen for
+    being used by the containers through the Docker
+    [host](https://docs.docker.com/network/host/) network.
+
+
+## Prerequisites
+
+To build and deploy the Docker images, `docker` and optionally `docker-compose` shall be available.
+The docker host should have access to the [Docker Hub](https://hub.docker.com/) and
+[CentOS](https://www.centos.org/) offical repositories.  Finally,
+[hugepages](https://www.kernel.org/doc/Documentation/vm/hugetlbpage.txt) linux kernel feature shall
+be enabled on the docker host.
+
+The platform was tested and validated with the [CentOS8](https://hub.docker.com/_/centos) official
+docker image.  However other RHEL-like distributions such as
+[rocky8.4](https://hub.docker.com/r/rockylinux/rockylinux) should be supported.
+
+!!! warning
+    Some distributions are not yet well supported such as
+    [rocky8.5](https://hub.docker.com/r/rockylinux/rockylinux): issue with the management of
+    hugepages with the [spdk](https://spdk.io/) library.
+
+
+## Building Docker Images
+
+### Building Base DAOS Image
+
+The first image to create is the `daos-base` image which is not intetended to be used as it, but as
+a base image for building the other three daos images.  This first image could be built directly
+from GitHub with the following command:
+
+```bash
+$ docker build --tag daos-base:centos8 \
+	"https://github.com/daos-stack/daos.git#master:utils/docker/vcluster/daos-base/el8"
```

Comment:
Have you tried this?  I get a submodule error when I run it here.

Reply:
I tested it with replacing the uri of my branch.
This last one will only work when the PR will be merged.

This should be replaced with
docker build --tag daos-base:centos8 \
	"https://github.com/knard-intel/daos.git#DEV/DAOS-9088:utils/docker/vcluster/daos-base/el8"

Reply:
I was using docker build "https://github.com/knard-intel/daos.git#DEV/DAOS-9088:utils/docker/vcluster/daos-base/el8"

and the error I get is

unable to prepare context: unable to 'git clone' to temporary context directory: error initializing submodules: usage: git submodule [--quiet] add [-b <branch>] [-f|--force] [--name <name>] [--reference <repository>] [--] <repository> [<path>]
   or: git submodule [--quiet] status [--cached] [--recursive] [--] [<path>...]
   or: git submodule [--quiet] init [--] [<path>...]
   or: git submodule [--quiet] deinit [-f|--force] [--] <path>...
   or: git submodule [--quiet] update [--init] [--remote] [-N|--no-fetch] [-f|--force] [--rebase] [--reference <repository>] [--merge] [--recursive] [--] [<path>...]
   or: git submodule [--quiet] summary [--cached|--files] [--summary-limit <n>] [commit] [--] [<path>...]
   or: git submodule [--quiet] foreach [--recursive] <command>
   or: git submodule [--quiet] sync [--recursive] [--] [<path>...]
: exit status 1

Reply:
Do not understand, I tries from my personal computer at home and it is working.
Which version of docker and git do you have ?

From my side , I tried with the following version of docker: Docker version 20.10.11, build dea9396
And the following version of git: git version 2.25.1 and git version 2.27.0